proposals/0019-graphql.md

```diff
@@ -76,7 +81,320 @@ The result of this query is a `GraphQLResult` that encodes to the following JSON
 { "hello": "world" }
 ```
 
-For more in-depth examples, see the [repository readme](https://github.com/GraphQLSwift/GraphQL/blob/master/README.md)
+### Swift Type Usage
+
+Graphiti includes support for using Swift types in the schema itself. To connect the Swift type with the GraphQL one, include a `Type` block in the API declaration, composed of `Field`s. For example, we can integrate a `Person` object into the API:
+
+```swift
+import Graphiti
+import NIO
+
+let eventLoopGroup = MultiThreadedEventLoopGroup(numberOfThreads: 1)
+
+struct Person: Codable {
+    let name: String
+    let age: Int
+    let height: Float
+}
+
+let characters = [
+    Person(name: "Johnny Utah", age: 23, height: 1.85),
+    Person(name: "Bodhi", age: 27, height: 1.8),
+]
+
+struct PersonResolver {
+    func people(context: NoContext, arguments: NoArguments) -> [Person] {
+        return characters
+    }
+}
+
+struct PersonAPI : API {
+    typealias ContextType = NoContext
+    let resolver = PersonResolver()
+    let schema = try! Schema<PersonResolver, NoContext> {
+        Type(Person.self) {
+            Field("name", at: \.name)
+            Field("age", at: \.age)
+            Field("height", at: \.height)
+        }
+        Query {
+            Field("people", at: PersonResolver.people)
+        }
+    }
+}
+
+let result = try await PersonAPI().execute(
+    request: """
+    {
+      people {
+        name
+        age
+      }
+    }
+    """,
+    context: NoContext(),
+    on: eventLoopGroup
+)
+```
+
+The `result` above could be decoded to a JSON of the form:
+
+```json
+{
+  "people": [
+    "name": "Johnny Utah",
+    "name": "Bodhi"
```

Comment:
Aee these suppose to be objects?

{"name": "J", "age": 6},
{"name": "B", "age": 20}